### ISSUE TITLE
EarlyStopping の初期化を setup_training() に統一する

### ISSUE THREAD

**Issue by kurorosu** [kurorosu/pochitrain#131]
## 概要

`PochiTrainer` の訓練コンポーネント初期化に一貫性の問題がある. `CheckpointStore`, `Evaluator`, `TrainingConfigurator` は `__init__` または `setup_training()` で初期化されるのに対し, `EarlyStopping` だけが `train()` メソッド内部で初期化されている.

### 初期化タイミングの不一致

| コンポーネント | 初期化場所 |
|---|---|
| CheckpointStore | `__init__` |
| Evaluator | `__init__` |
| TrainingConfigurator | `__init__` |
| optimizer / scheduler / criterion | `setup_training()` |
| MetricsTracker | `train()` 内 (ローカル変数, 問題なし) |
| **EarlyStopping** | **`train()` 内 (インスタンス変数)** |

`EarlyStopping` は CLI (`pochi.py:249-251`) から `trainer.early_stopping_config` に dict を直接代入し, `train()` の冒頭で `EarlyStopping` インスタンスを生成している.

```python
# pochi.py (CLI側)
trainer.early_stopping_config = early_stopping_config

# pochi_trainer.py train() 内 (L276-290)
if self.early_stopping_config is not None:
    es_config = self.early_stopping_config
    if es_config.get("enabled", False):
        self.early_stopping = EarlyStopping(
            patience=es_config.get("patience", 10),
            ...
        )
```

### 問題点

1. **一貫性の欠如**: 他の訓練コンポーネントは `setup_training()` で構築されるのに, `EarlyStopping` だけ `train()` で初期化される
2. **生のdictによる設定注入**: `early_stopping_config` が型なし dict としてインスタンス変数に代入され, `train()` 内で dict から手動パースされる
3. **初期化の分散**: `setup_training()` を呼んだ時点では訓練設定が「完了」したように見えるが, 実際には `EarlyStopping` が未設定

## 修正方針

`setup_training()` に `early_stopping_config` パラメータを追加し, `EarlyStopping` の初期化を `setup_training()` に統一する.

```python
# pochitrain/pochi_trainer.py
def setup_training(
    self,
    learning_rate: float = 0.001,
    optimizer_name: str = "Adam",
    ...
    early_stopping_config: Optional[Dict[str, Any]] = None,  # 追加
) -> None:
    # 既存の TrainingConfigurator による構築
    components = self.training_configurator.configure(...)
    ...

    # EarlyStopping の初期化 (train() から移動)
    if early_stopping_config and early_stopping_config.get("enabled", False):
        self.early_stopping = EarlyStopping(
            patience=early_stopping_config.get("patience", 10),
            min_delta=early_stopping_config.get("min_delta", 0.0),
            monitor=early_stopping_config.get("monitor", "val_accuracy"),
            logger=self.logger,
        )
```

```python
# pochitrain/cli/pochi.py (変更後)
trainer.setup_training(
    ...
    early_stopping_config=config.get("early_stopping"),
)
# trainer.early_stopping_config = ... の直接代入を削除
```

### 削除対象

- `PochiTrainer.early_stopping_config` インスタンス変数 (dict)
- `train()` 内の `EarlyStopping` 初期化ブロック (L276-290)

## 対象ファイル

- `pochitrain/pochi_trainer.py` - `setup_training()` に EarlyStopping 初期化を移動, `train()` から削除
- `pochitrain/cli/pochi.py` - `setup_training()` 経由で設定を渡すように変更
- `tests/` - 対応するテストの修正

## Labels

`refactor`

## Branch

`refactor/early-stopping-init`
